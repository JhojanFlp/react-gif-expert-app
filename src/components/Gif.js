import React from 'react'
import PropTypes from 'prop-types'

const Gif = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__backInUp">
      <img src={url} alt={title} />
      <p>{title || id}</p>
    </div>
  ) 
}

Gif.propTypes = {
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
}

export default Gif
