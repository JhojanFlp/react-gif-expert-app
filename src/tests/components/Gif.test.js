import React from 'react';
import { shallow } from 'enzyme';

import Gif from '../../components/Gif';

describe('Test in Gif.js', () => {
  const props = { id: '1', url: 'https://example.com', title: 'TÃ­tulo' };
  const wrapper = shallow(<Gif {...props} />);
  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  })
  test('should title prop in a p tag', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(props.title || props.id);
  })
  test('should img with props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(props.url);
    expect(img.prop('alt')).toBe(props.title)
  })
  test('should class animate__backInUp', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(div.hasClass('animate__backInUp')).toBe(true)
    expect(className.includes('animate__backInUp')).toBe(true);
  })
  
})
