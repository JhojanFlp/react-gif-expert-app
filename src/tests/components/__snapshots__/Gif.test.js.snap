// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Gif.js should render component 1`] = `
<div
  className="card animate__animated animate__backInUp"
>
  <img
    alt="Título"
    src="https://example.com"
  />
  <p>
    Título
  </p>
</div>
`;
